src/models/tiles/geometry/geometry-content.ts

```diff
@@ -1251,6 +1269,25 @@ export const GeometryContentModel = GeometryBaseContentModel
       return elems ? elems as JXG.GeometryElement[] : undefined;
     }
 
+    function setSelectedColor (color: number, board: JXG.Board) {
+      self.selectedColor = color;
+      const selectedIds = self.getSelectedIds(board);
+      selectedIds.forEach(id => {
+        const obj = self.getObject(id);
+        if (isPolygonModel(obj) || isPointModel(obj) || isCircleModel(obj)) {
+          obj.setColorScheme(color);
+          const change: JXGChange = {
+            operation: "update",
+            target: "object",
+            targetID: id,
+            properties: { colorScheme: color }
+          };
+          applyAndLogChange(board, change);
```

Comment:
Note that this (`applyAndLogChange`) will generate an event log entry for each object that is updated - 5 log entries if I change the color of a square.

I think we'd rather have just one log message.  So inside the loop, you can just compile a list of the IDs that have been changed.  Then call `applyAndLogChange` just once, passing the list of IDs in the `targetID` field.  I think that should do the right thing.